cmd/saga/worker/app/worker/worker.go

```diff
@@ -70,7 +72,21 @@ func NewConnToProfile() (*grpc.ClientConn, error) {
 func dial(addr string) (*grpc.ClientConn, error) {
 	return grpc.Dial(addr,
 		grpc.WithTransportCredentials(insecure.NewCredentials()),
-		grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor()),
+		grpc.WithChainUnaryInterceptor(
+			// interceptor to extract Baggage from workflow.Context and inject into new context
+			func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+				bg, ok := ctx.Value(ctxpropagation.PropagateKey).(baggage.Baggage)
+
+				if ok {
+					// INJECT!
+					ctx = baggage.ContextWithBaggage(ctx, bg)
+				}
+
+				return invoker(ctx, method, req, reply, cc, opts...)
+			},
+			// create span, do context propagation
+			otelgrpc.UnaryClientInterceptor(),
+		),
```

Comment:
Here we are in the Temporal worker. Before we invoke the next interceptor in the chain, we're going to pull baggage from context using our special `PropagateKey` and then re-set it in context.